### DISCUSSION TITLE
return back!

### DISCUSSION THREAD

**Discussion by PureDreamer** [fastapi/fastapi#2649] [Questions] [answered]
Hey there Im using fastapi-mail to send emails in the background.
I wondered is there any way I can return to the page I came from instead of getting JSONresponse?

A little bit of info:
from the home page I'm calling a form that sends me to another route. A route which is only for sending email.
if it is correct it return JSONresponse

I want it to get me back to the homepage (but not hard coded)

**Comment by Mause** [comment, marked as answer]
You could redirect them to the value of the referer header?

**Comment by krishnardt** [comment]
you can use RedirectResponse

from fastapi.responses import RedirectResponse
RedirectResponse(url="use the url here to which you want to redirect to", status_code=status.HTTP_303_SEE_OTHER)


edit1:
as @Mause said, there will be reference in the requests... you can fetch them and use it on RedirectResponse.

**Comment by PureDreamer** [comment]
thanks for the answer!
Second one if I may...
Is there a way to call a function which is not a route inside jinja?

**Reply by krishnardt** [reply]
jinja is for dynamic templates and not for functions as per my view.
Can you elaborate the question?
